test/page-objects/communityMarketPage.js

```diff
@@ -1,96 +1,88 @@
+    this.rarityCheckbox = (rarityType) => new Label(`//label[contains(@for, 'Rarity_${rarityType}')]`, "Rarity Checkbox");
 
-    this.searchResultsTable = new Label("//div[@id='searchResultsRows']", "Search Results Table")
-    this.firstSearchItem = new Label("//a[@id='resultlink_0']", "First Search Item")
-    this.searchTags = new Label(`//div[contains(@class, 'search_results_header')]//div`, "Search Tags")
-    this.sortPriceButton = new Button("//div[@data-sorttype='price']", "Sort Price Button")
+    this.searchResultsTable = new Label("//div[@id='searchResultsRows']", "Search Results Table");
+    this.firstSearchItem = new Label("//a[@id='resultlink_0']", "First Search Item");
+    this.searchTags = new Label("//div[contains(@class, 'search_results_header')]//div", "Search Tags");
+    this.sortPriceButton = new Button("//div[@data-sorttype='price']", "Sort Price Button");
   }
 
   async showAdvancedOptions() {
-    this.advancedOptionsButton.click()
+    await this.advancedOptionsButton.click();
   }
 
-  async isAdvancedOptionsWindowOpened() {
-    return await this.advancedSearchWindow.state().waitForDisplayed()
+  isAdvancedOptionsWindowOpened() {
+    return this.advancedSearchWindow.state().waitForDisplayed();
   }
 
   async selectGameForSearch(gameName) {
-    await this.gameSelect.click()
-    await this.gameOption(gameName).moveTo()
-    await this.gameOption(gameName).click()
+    await this.gameSelect.click();
+    await this.gameOption(gameName).moveTo();
+    await this.gameOption(gameName).click();
   }
 
   async selectHeroForSearch(heroName) {
-    await this.heroSelect.selectOptionByText(heroName)
+    await this.heroSelect.selectOptionByText(heroName);
   }
 
   async selectRarityForSearch(rarityType) {
-    await this.rarityCheckbox(rarityType).click()
+    await this.rarityCheckbox(rarityType).click();
   }
 
   async clickSearchButton() {
-    await this.searchButton.click()
+    await this.searchButton.click();
   }
 
-  async isSearchResultsTableDisplayed() {
-    return await this.searchResultsTable.state().isDisplayed()
+  isSearchResultsTableDisplayed() {
+    return this.searchResultsTable.state().isDisplayed();
   }
 
-  async areSearchTagsCorrect(expectedTags) {
-    const tagElementsArray = await this.searchTags.findAll(Label, "//a", "searchTag")
-    const searchTagArray = []
+  async getSearchTags() {
+    const tagElementsArray = await this.searchTags.findAll(Label, "//a", "searchTag");
+    const searchTagsArray = tagElementsArray.slice(0, -1).map(async (tagElement) => await tagElement.getText());
 
-    for await (let tagElement of tagElementsArray) {
-      const searchTagText = await tagElement.getText()
-      searchTagArray.push(searchTagText)
-    }
-
-    for (const tag of expectedTags) {
-      if (!searchTagArray.includes(tag)) {
-        return false
-      }
-    }
-
-    return true
+    return await Promise.all(searchTagsArray);
```

Comment:
redundant await on a return statement
https://stackoverflow.com/questions/60842265/why-am-i-not-getting-redundant-return-await-eslint-no-return-await-warning

Reply:
Are you sure, that searchTagsArray is an array of promises?

Reply:
Fixed. Yes, its an array of promises